### PR TITLE
lib:simulations:hypocycloidcurve:Hypocycloid Curve implemented

### LINKED ISSUE #164: simulation: Hypocycloid Curve
**Hypocycloid Curve**
[https://en.m.wikipedia.org/wiki/Hypocycloid](https://en.m.wikipedia.org/wiki/Hypocycloid)

**Category Proposed:** Mathematics

**Additional points/links:**

### PR DESCRIPTION
**Pull request by Snehaa1203**
fixed #164

### CODE REVIEW

File: lib/src/data/simulations.dart [hunk 1]
```diff
@@ -11,6 +11,7 @@ import 'package:shared_preferences/shared_preferences.dart';
 import 'package:simulate/src/simulations/lissajous_curve.dart';
 import 'package:simulate/src/simulations/epicycloid_curve.dart';
 import 'package:simulate/src/simulations/maurer_rose.dart';
+import  'package:simulate/src/simulations/Hypocycloid.dart';
 
 class Simulations with ChangeNotifier {
   static var _favorites = [-1, -1, -1, -1, -1, -1, -1, -1, -1];
```
Comment (yashugarg) [inline lib/src/data/simulations.dart]: add another "-1" to _favorites array
and a "9" to the _mathematics arra

File: lib/src/data/simulations.dart [hunk 2]
```diff
@@ -141,6 +143,16 @@ class Simulations with ChangeNotifier {
         infoLink: 'https://en.wikipedia.org/wiki/Maurer_rose',
         fav: _favorites[8],
       ),
+      SimulationCard(
+        id: 9,
+        simulationName: 'Hypocycloid curve',
+        image: theme.darkTheme
+            ? 'assets/simulations/Epicycloid1Dark.png'
```
Comment (yashugarg) [inline lib/src/data/simulations.dart]: put the right images associated with your simulation

File: lib/src/simulations/Hypocycloid.dart [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+        }
+        canvas.drawPoints(PointMode.polygon, points, paint);
+      }
+    } else {
+      Paint paint = new Paint();
+      paint.style = PaintingStyle.stroke;
+      paint.strokeWidth = 2;
+      paint.color = Theme.of(context).accentColor;
+      canvas.drawCircle(coor, innerRadius.toDouble(), paint);
+      smallCenter = Offset((innerRadius + outerRadius) * cos(time),
```
Comment (yashugarg) [inline lib/src/simulations/Hypocycloid.dart]: make it 
innerRadius - outerRadius
to display the rotating circle inside the base circle

**Review by yashugarg — CHANGES_REQUESTED** [review]
the simulation works fine
great work
just do these small changes

**Review by Snehaa1203 — COMMENTED** [review]
I have made the changes.
Please accept the PR

**Review by yashugarg — APPROVED** [review]

**Pull request merged**